packet.cpp

```diff
@@ -0,0 +1,133 @@
+#include "packet.h"
+
+Packet::Packet(char data[24], tcp::socket& socket,
+               const std::shared_ptr<KeyValueStore>& k)
```

Comment:
I have become a fan of putting the comma at the beginning of the line:

```c++
    : bod_len(readUInt32LE(data, 8))
    , opaque(readUInt32LE(data, 12))
    , key_len(readUInt16LE(data, 2))
    , op(data[1])
    , ext_len(data[4]) 
{ }
```
It looks a tad more uniform.